features/exploredash/src/main/java/org/dash/wallet/features/exploredash/data/MerchantDao.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 Dash Core Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dash.wallet.features.exploredash.data
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import kotlinx.coroutines.flow.Flow
+import org.dash.wallet.features.exploredash.data.model.Merchant
+import org.dash.wallet.features.exploredash.data.model.MerchantFTS
+
+@Dao
+interface MerchantDao {
```

Comment:
Sorry, I didn't quite get it, MerchantRepository is for interacting with Firebase database, what does dao has to do with it?
Suspend functions in Dao exist so that you can call them from any context (including Main) and the don't block, because Room runs them in IO internally.

Reply:
Ok, I get what's the reason for this confusion. Let's discuss this in slack.